### PR TITLE
Add check for participant ID

### PR DESCRIPTION
**Pull request by danlim715**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- Add a check for participant ID before calling the ITF async job.
- Pension benefit to own this change

## Related issue(s)

- [*Link to ticket created in va.gov-team repo*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88635)

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v0/in_progress_forms_controller.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ def update
       form.update!(form_data: params[:form_data] || params[:formData], metadata: params[:metadata])
 
       if Flipper.enabled?(:intent_to_file_lighthouse_enabled) && form.id_previously_changed? &&
-         Lighthouse::CreateIntentToFileJob::ITF_FORMS.include?(form.form_id)
+         Lighthouse::CreateIntentToFileJob::ITF_FORMS.include?(form.form_id) && @current_user.participant_id.present?
```
Comment (mjknight50) [inline app/controllers/v0/in_progress_forms_controller.rb:29]: Should this be?
@current_user?.participant_id?.present?

My rails syntax is bad, but this is more JS style.

Comment (danlim715) [reply app/controllers/v0/in_progress_forms_controller.rb:29]: u mean adding safe nav like this?: `@current_user&.participant_id&.present?`. That would protect against a nil @current_user which is not possible at this point when creating a new in progress form.

Comment (mjknight50) [reply app/controllers/v0/in_progress_forms_controller.rb:29]: Sounds good.  Thanks!

**Review by mjknight50 — APPROVED** [review]
GTG

**Review by stevenjcumming — APPROVED** [review]
The conditional is a bit of an eye sore, but maybe a future refactor-er will find a clean way to present that.

**Pull request merged**